src/flow/src/adapter.rs

```diff
@@ -14,19 +14,678 @@
 
 //! for getting data from source and sending results to sink
 //! and communicating with other parts of the database
+#![warn(unused_imports)]
+
+use std::collections::{BTreeMap, HashMap};
+use std::sync::Arc;
+
+use api::v1::{RowDeleteRequest, RowDeleteRequests, RowInsertRequest, RowInsertRequests};
+use catalog::kvbackend::KvBackendCatalogManager;
+use common_base::Plugins;
+use common_error::ext::BoxedError;
+use common_frontend::handler::FrontendInvoker;
+use common_meta::key::TableMetadataManagerRef;
+use common_meta::kv_backend::KvBackendRef;
+use common_runtime::JoinHandle;
+use common_telemetry::{debug, info};
+use datatypes::schema::ColumnSchema;
+use datatypes::value::Value;
+use greptime_proto::v1;
+use itertools::Itertools;
+use minstant::Anchor;
+use query::{QueryEngine, QueryEngineFactory};
+use serde::{Deserialize, Serialize};
+use session::context::QueryContext;
+use snafu::{OptionExt, ResultExt};
+use store_api::storage::{ConcreteDataType, RegionId};
+use table::metadata::TableId;
+use tokio::sync::{oneshot, watch, Mutex, RwLock};
+
+use crate::adapter::error::{ExternalSnafu, TableNotFoundSnafu, UnexpectedSnafu};
+pub(crate) use crate::adapter::node_context::FlownodeContext;
+use crate::adapter::parse_expr::parse_fixed;
+use crate::adapter::table_source::TableSource;
+use crate::adapter::util::column_schemas_to_proto;
+use crate::adapter::worker::{create_worker, Worker, WorkerHandle};
+use crate::compute::ErrCollector;
+use crate::expr::GlobalId;
+use crate::repr::{self, DiffRow, Row};
+use crate::transform::sql_to_flow_plan;
 
 pub(crate) mod error;
-pub(crate) mod node_context;
-mod table_source;
+mod flownode_impl;
+mod parse_expr;
+mod server;
+#[cfg(test)]
+mod tests;
 mod util;
+mod worker;
 
-pub(crate) use node_context::FlownodeContext;
-pub(crate) use table_source::TableSource;
+pub(crate) mod node_context;
+mod table_source;
 
-mod worker;
+use error::Error;
 
 pub const PER_REQ_MAX_ROW_CNT: usize = 8192;
+
 // TODO: refactor common types for flow to a separate module
 /// FlowId is a unique identifier for a flow task
-pub type FlowId = u32;
+pub type FlowId = u64;
 pub type TableName = [String; 3];
+
+/// Options for flow node
+#[derive(Clone, Default, Debug, Serialize, Deserialize)]
+#[serde(default)]
+pub struct FlownodeOptions {
+    /// rpc address
+    pub rpc_addr: String,
+}
+
+/// Flownode Builder
+pub struct FlownodeBuilder {
+    opts: FlownodeOptions,
+    plugins: Plugins,
+    kv_backend: Option<KvBackendRef>,
+    table_meta: TableMetadataManagerRef,
+    catalog_manager: Arc<KvBackendCatalogManager>,
```

Comment:
It's better to avoid using `KvBackendCatalogManager` directly since it's just an implementation of the catalog manager. We should use `CatalogManagerRef` instead.